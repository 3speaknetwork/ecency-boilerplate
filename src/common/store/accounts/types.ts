import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;