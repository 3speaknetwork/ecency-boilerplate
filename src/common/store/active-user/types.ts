import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
