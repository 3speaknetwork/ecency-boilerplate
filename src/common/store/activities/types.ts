interface BaseActivity {
    event: "notify";
    source: string;
    target: string;
    timestamp: string;
    extra: any;
}

export interface VoteActivity extends BaseActivity {
    type: "vote",
    extra: {
        permlink: string;
        weight: number;
        title: string | null;
        img_url: string | null;
    }
}

export interface MentionActivity extends BaseActivity {
    type: "mention",
    extra: {
        permlink: string;
        is_post: 0 | 1;
        title: string | null;
        img_url: string | null;
    }
}

export interface FollowActivity extends BaseActivity {
    type: "follow",
    extra: {
        what: string[]
    }
}

export interface ReplyActivity extends BaseActivity {
    type: "reply",
    extra: {
        title: string;
        body: string;
        json_metadata: string;
        permlink: string;
        parent_author: string;
        parent_permlink: string;
        parent_title: string | null;
        parent_img_url: string | null;
    }
}

export interface ReblogActivity extends BaseActivity {
    type: "reblog";
    extra: {
        permlink: string;
        title: string | null;
        img_url: string | null;
    }

}

export interface TransferActivity extends BaseActivity {
    type: "transfer";
    extra: {
        amount: string;
        memo: string;
    }
}


export type Activity = VoteActivity | MentionActivity | FollowActivity | ReplyActivity | ReblogActivity | TransferActivity;
