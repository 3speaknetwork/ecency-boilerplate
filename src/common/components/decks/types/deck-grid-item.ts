interface BaseDeckGridItem {
  key: number;
  type: "u" | "ac" | "co" | "w" | "n" | "tr" | "to" | "s" | "cu";
  settings: unknown;
}

export interface UserDeckGridItem extends BaseDeckGridItem {
  type: "u";
  settings: {
    username: string;
    contentType: string;
  };
}

export type DeckGridItem = BaseDeckGridItem | UserDeckGridItem;
