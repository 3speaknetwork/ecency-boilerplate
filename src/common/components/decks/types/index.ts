export * from "./deck-grid-item";
export * from "./deck-grid";
