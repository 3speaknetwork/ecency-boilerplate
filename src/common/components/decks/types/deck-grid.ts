import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  columns: DeckGridItem[];
}
