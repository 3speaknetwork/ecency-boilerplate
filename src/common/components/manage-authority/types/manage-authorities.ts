import { Account } from "../../../store/accounts/types";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting"
};

export interface PublicKeys {
  publicOwnerKey?: any;
  publicActiveKey?: any;
  publicPostingKey?: any;
}

export interface AccountDataType {
  postingsAuthority: Array<any>;
  posting: Array<any>;
  owner: Array<any>;
  active: Array<any>;
  weight: number;
  memokey: string;
  account: Account;
  PublicKeys: {
    publicOwnerKey: any;
    publicActiveKey: any;
    publicPostingKey: any;
  };
}

export const actionType = {
  Revoke: "revoke"
};
