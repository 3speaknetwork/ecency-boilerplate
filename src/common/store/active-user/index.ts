import { Dispatch } from "redux";

import { Account } from "../accounts/types";
import { ActiveUser, Actions, ReloadAction, UpdateAction, ActionTypes } from "./types";

import * as ls from "../../util/local-storage";

const load = (): ActiveUser | null => {
  const name = ls.get("active_user");
  if (name && ls.get(`user_${name}`)) {
    return {
      username: name,
      data: { name },
    };
  }

  return null;
};

export const initialState: ActiveUser | null = typeof window !== "undefined" ? load() : null;

export default (state: ActiveUser | null = initialState, action: Actions): ActiveUser | null => {
  switch (action.type) {
    case ActionTypes.RELOAD: {
      return load();
    }
    case ActionTypes.UPDATE: {
      const { data } = action;
      return Object.assign({}, state, { data });
    }
    default:
      return state;
  }
};

/* Actions */
export const setActiveUser = (name: string | null) => async (dispatch: Dispatch) => {
  if (name) {
    ls.set("active_user", name);
  } else {
    ls.remove("active_user");
  }

  dispatch(reloadAct());
};

export const updateActiveUser = (data: Account) => async (dispatch: Dispatch) => {
  dispatch(updateAct(data));
};

/* Action Creators */
export const reloadAct = (): ReloadAction => {
  return {
    type: ActionTypes.RELOAD,
  };
};

export const updateAct = (data: Account): UpdateAction => {
  return {
    type: ActionTypes.UPDATE,
    data,
  };
};
