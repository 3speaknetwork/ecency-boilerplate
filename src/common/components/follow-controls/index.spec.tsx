import React from 'react';
import renderer from "react-test-renderer";

import {ActiveUser} from "../../store/active-user/types";
import FollowControls from './index';

const allOver = () => new Promise((resolve) => setImmediate(resolve));

let MOCK_MODE: number = 0;

jest.mock("../../api/hive", () => ({
    getFollowing: (follower: string,
                   startFollowing: string,
                   followType: string,
                   limit: number) =>
        new Promise((resolve) => {

            if (MOCK_MODE === 1) {
                resolve([]);
                return;
            }

            if (MOCK_MODE === 2) {
                resolve([{
                    follower: 'fooo1',
                    following: 'bar'
                }]);
                return;
            }

            if (MOCK_MODE === 3) {
                if (followType === 'ignore') {
                    resolve([{
                        follower: 'fooo1',
                        following: 'bar'
                    }]);

                } else {
                    resolve([]);
                }
                return;
            }
        }),
}));

const defProps = {
    users: [],
    activeUser: null,
    targetUsername: 'bar',
    setActiveUser: () => {
    },
    updateActiveUser: () => {
    },
    deleteUser: () => {
    }
};

describe('FollowControls', () => {
    it('(1) Default render. No active user.', () => {
        const props = {...defProps};

        const component = renderer.create(<FollowControls {...props}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('(2) Default Render with active account.', async () => {
        MOCK_MODE = 1;

        const activeUser: ActiveUser = {
            username: 'fooo1',
            data: {name: 'fooo1'}
        }

        const props = {
            ...defProps,
            activeUser
        };

        const component = renderer.create(<FollowControls {...props}/>);
        await allOver();
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('(2) Following.', async () => {
        MOCK_MODE = 2;

        const activeUser: ActiveUser = {
            username: 'fooo1',
            data: {name: 'fooo1'}
        }

        const props = {
            ...defProps,
            activeUser
        };

        const component = renderer.create(<FollowControls {...props}/>);
        await allOver();
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('(2) Muted.', async () => {
        MOCK_MODE = 3;

        const activeUser: ActiveUser = {
            username: 'fooo1',
            data: {name: 'fooo1'}
        }

        const props = {
            ...defProps,
            activeUser
        };

        const component = renderer.create(<FollowControls {...props}/>);
        await allOver();
        expect(component.toJSON()).toMatchSnapshot();
    });
});
