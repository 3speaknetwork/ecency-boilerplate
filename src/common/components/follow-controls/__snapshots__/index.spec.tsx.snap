// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowControls (1) Default render. No active user. 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "5px",
      }
    }
    type="button"
  >
    Follow
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Mute
  </button>
</div>
`;

exports[`FollowControls (2) Default Render with active account. 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "5px",
      }
    }
    type="button"
  >
    Follow
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Mute
  </button>
</div>
`;

exports[`FollowControls (2) Following. 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "5px",
      }
    }
    type="button"
  >
    Unfollow
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Mute
  </button>
</div>
`;

exports[`FollowControls (2) Muted. 1`] = `
<div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "5px",
      }
    }
    type="button"
  >
    Follow
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Unmute
  </button>
</div>
`;
